code/modules/power/supermatter/supermatter.dm

```diff
@@ -235,7 +237,7 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 
 /obj/machinery/power/supermatter_crystal/process_atmos()
 	// PART 1: PRELIMINARIES
-	if(disable_process)
+	if(disable_process || time_frozen)
```

Comment:
If you are just going to replicate what disable process does, consider making it a var that can hold 3 values (defines):

- Not disabled
- Disabled in game (if check in your unfreeze supermatter code, set it back to none if its this)
- Disabled fully (unfreeze supermatter cant unfreeze it if its set to this, set hugbox supermatters to this)

You can look into the defines i use in sm delam priority for an example.



If you make a new variable for every single little feature sm will get ugly again :(